crates/evm/fuzz/src/strategies/state.rs

```diff
@@ -371,10 +371,17 @@ impl FuzzDictionary {
     }
 
     pub fn revert(&mut self) {
-        // Revert new values collected during the run.
+        // Revert values and addresses collected during the run, using recorded indexes.
+        // Indexes are sorted and removed in reverse in order to point to same value as when they
+        // were recorded.
+        self.new_values.sort();
+        self.new_values.reverse();
```

Comment:
`sort_unstable` + `iter().rev()` should be better

Reply:
technically they should already be sorted but it should be ok to just do it again

Reply:
@DaniPopes please check https://github.com/foundry-rs/foundry/pull/8080/commits/8857b105288d8725b5601f99afb974b6d8b97501, it basically retains the size of values and addresses inserted from db and at the end removes all the other values from that index to end, in reverse order. Also got rid of `collected` flag since the duped values are no longer used.

Reply:
Great! You can simplify it by calling [`IndexSet::truncate`](https://docs.rs/indexmap/latest/indexmap/set/struct.IndexSet.html#method.truncate)

Reply:
ah, nice! missed that one. changed in https://github.com/foundry-rs/foundry/pull/8080/commits/6448d34d5ff98228cf4cd0669477541761bacba4